core/src/main/java/org/axonframework/config/DefaultConfigurer.java

```diff
@@ -227,6 +232,15 @@ protected ParameterResolverFactory defaultParameterResolverFactory(Configuration
                                                      new ConfigurationParameterResolverFactory(config));
     }
 
+    /**
+     * Provides the default HandlerDefinition. Subclasses may override this method to provide their own default.
+     *
+     * @return The default HandlerDefinition to use
+     */
+    protected HandlerDefinition defaultHandlerDefinition() {
+        return new MultiHandlerDefinition(new ClasspathHandlerDefinition(getClass().getClassLoader()));
```

Comment:
Is there a specific reason to wrap the `ClasspathHandlerDefinition` in a `MultiHandlerDefinition`?

When continuing the review, I noticed this (probably) has to do with the Handler Enhancers. Changing the `ClasspathHandlerDefinition`'s constructor to a static factory method will resolve this issue.